Alleviate package-lock changing problems

ericcornelissen opened the pull request:
Following up on #92, #104, and https://github.com/simple-icons/simple-icons/pull/6179 this is an attempt to alleviate some of the trouble with `package-lock.json` version 1 and 2 (aka npm v6 or lower and npm v7 or higher).

1. First, I specified the desired NPM version (and by extension the desired `package-lock.json` version) in this project's manifest file. (Also fixing https://github.com/simple-icons/simple-icons/pull/6349 for this repo.)
2. Then, I set `engine-strict=true` in a newly added `.npmrc` file for this project. This will make `npm` error if your Node or NPM version does not comply with the engines values of this project or any of our dependencies.
3. Lastly, I added an `.nvmrc` file to allow users of [NVM](https://github.com/nvm-sh/nvm) (which I highly encourage you to pick up you're not using it yet) (also [for Windows](https://github.com/coreybutler/nvm-windows)) to easily switch to the correct NodeJS & npm version for this project.

~~There is one more thing I can think of that we can do, which is to add a `Dockerfile` for development. I'll leave that for a separate Pull Request though, if it is something we want to add.~~ I just realized, [we already have this](https://github.com/simple-icons/simple-icons-website/blob/master/Dockerfile)! And [it's even documented](https://github.com/simple-icons/simple-icons-website/blob/master/CONTRIBUTING.md#using-docker)!

If this works well, we can apply (some of) these changes to other simple-icons projects as well :slightly_smiling_face: 

---

cc @LitoMore & @dylvaz

ericcornelissen committed:
Set allowed npm version in manifest's engines

- Fix the name of the engines field
- Specify the suported npm version range

ericcornelissen committed:
Set `eninge-instrict=true` in .npmrc

This will force anyone using npm commands in this project to adhere to
the values of the "engines" field of this project (and dependencies)
(unless they use --force).

ericcornelissen committed:
Add an `.nvmrc` file

To make it easier for contributors that use NVM to switch to the right
NodeJS & npm version.

LitoMore commented on package.json:
```diff
@@ -61,7 +61,8 @@
-    "node": ">=12.16.0"
+  "engines": {
+    "node": ">=12.16.0",
+    "npm": "<7.0.0"
```
The npm is always binded with Node.js. This requirement will be a little strange. We could use `"node": "^12.16.0",` instead of `>=12.16.0`.

But since npm v8 released. We better not use npm 6, IMO. Developers should keep their npm up to date.

FYI: https://github.com/npm/cli

ericcornelissen replied:
> The npm is always binded with Node.js. This requirement will be a little strange. We could use `"node": "^12.16.0",` instead of `>=12.16.0`.

In that case it should be `"node": "^12.16.0 || ^14.18.0 || ^16.11.0",` as `^12.16.0` only allows using major release v12. I don't see a problem with allowing users to use npm `<7.0.0` with a node version that comes with npm `>7.0.0` since you can downgrade npm if you want (that might be easier for some people than switching to different node version).

> But since npm v8 released. We better not use npm 6, IMO. Developers should keep their npm up to date.

I'd be happy to switch to package-lock v2 in all projects except the main simple-icons project itself given there's a low volume of contributions here - provided other @simple-icons/maintainers agree. If we do change this to `"npm": ">7.0.0",` then I'd suggest we also change `.nvmrc` to v16.

In the main project I'd be more cautious as contributions so far have shown that a lot of contributors are still on npm `<7.0.0` (provided we adopt this strategy in that project of course).

As a last note, I'm not completely in agreement here and until npm v6 is officially deprecated I'd be fine using it. Upgrading, at least to v7 or v8 is as simple as running `npm install` and committing the new `package-lock.json`. As far as a I know the primary advantage of the new lockfile version is efficiency.

LitoMore replied:
Good points. Let's move forward.

ericcornelissen committed:
Merge branch 'master' into npm-version-problems

ericcornelissen committed:
Merge branch 'master' into npm-version-problems

ericcornelissen committed:
Merge branch 'master' into npm-version-problems

ericcornelissen committed:
Set lockfile-version=1 in .npmrc

This tells npm version 8.1.0 and higher to use lockfile-version 1
rather than 2. In practice, it seems 8.1.0 and 8.1.1 have some problems
recognizing the option when it is specified in the .npmrc file, hence I
allowed npm versions higher than 8.1.2 to run `npm install` (in addition
to <7.0.0).
